### PR TITLE
Add platforms to user object in global state. State of platform switc…

### PR DESCRIPTION
**Pull request by nalanart**
…hes is persisted

### CODE REVIEW

File: client/src/hooks/getMentions.js [hunk 1]
```diff
@@ -1,9 +1,21 @@
...
+  let i = 0;
+  for (const platform in platforms) {
+    if (platforms[platform] === true) {
+      platformsArray.push(Object.keys(platforms)[i]);
+    }
+    i++;
+  }
+
+  let platformsString = platformsArray.join();
+
```
Comment (nalanart) [inline client/src/hooks/getMentions.js]: The best way I found to convert the keys from the platforms object into a string if it is true. Lmk if you know of a better method!

File: client/src/hooks/getMentions.js [hunk 2]
```diff
@@ -1,9 +1,21 @@
...
+export const getMentions = async (dispatch, keyword, platforms) => {
+  let platformsArray = [];
+
+  let i = 0;
+  for (const platform in platforms) {
+    if (platforms[platform] === true) {
+      platformsArray.push(Object.keys(platforms)[i]);
+    }
+    i++;
+  }
```
Comment (dproc96) [inline client/src/hooks/getMentions.js]: There's probably a cleaner way to do this, maybe try an Array.filter function

**Review by dproc96 — COMMENTED** [review]
One small suggestion to refactor, but otherwise this looks good!

**Review by nalanart — COMMENTED** [review]
> One small suggestion to refactor, but otherwise this looks good!

Thanks for the suggestion Dan!

**Pull request merged**